src/ray/gcs/redis_context.cc

```diff
@@ -560,10 +562,11 @@ Status ConnectRedisSentinel(RedisContext &context,
   }
 }
 
-std::vector<std::string> ResolveDNS(const std::string &address, int port) {
+std::vector<std::string> ResolveDNS(instrumented_io_context &io_service,
```

Comment:
If I'm calling ResolveDNS from the io_service thread, will this be a problem (deadlock)?

Reply:
Not a problem, my test script.
```
#include <boost/asio.hpp>
#include <boost/asio/ip/tcp.hpp>
#include <iostream>
#include <memory>

void resolveAddress(boost::asio::io_context& io_context, const std::string& host, const std::string& port) {
    try {
        // Create a resolver
        boost::asio::ip::tcp::resolver resolver(io_context);

        // Resolve the address
        boost::asio::ip::tcp::resolver::results_type endpoints = resolver.resolve(host, port);

        std::cout << "Resolved endpoints:" << std::endl;
        for (const auto& endpoint : endpoints) {
            std::cout << endpoint.endpoint() << std::endl;
        }
    } catch (const std::exception& e) {
        std::cerr << "Error: " << e.what() << std::endl;
    }
}

int main() {
    try {
        boost::asio::io_context io_context;

        // Post a functor into the io_context
        io_context.post([&io_context]() {
            std::cout << "Functor running in io_context." << std::endl;

            // Use the io_context with a TCP resolver
            resolveAddress(io_context, "example.com", "80");
        });

        // Run the io_context
        io_context.run();
    } catch (const std::exception& e) {
        std::cerr << "Exception: " << e.what() << std::endl;
    }

    return 0;
}
```

Reply:
Output for the above code snippet.
```
Functor running in io_context.
Resolved endpoints:
93.184.215.14:80
[2606:2800:21f:cb07:6820:80da:af6b:8b2c]:80
```